ocs_ci/utility/utils.py

```diff
@@ -2054,3 +2055,84 @@ def convert_device_size(unformatted_size, units_to_covert_to):
             return absolute_size / 1024
         elif units == 'Mi':
             return absolute_size
+
+
+def mirror_image(image):
+    """
+    Mirror image to mirror image registry.
+
+    Args:
+        image (str): image to be mirrored, can be defined just with name or
+                     with full url, with or without tag or digest
+
+    Returns:
+        str: the mirrored image link
+
+    """
+    # load pull-secret file to pull_secret dict
+    pull_secret_path = os.path.join(
+        constants.TOP_DIR,
+        "data",
+        "pull-secret"
+    )
+    with open(pull_secret_path) as pull_secret_fo:
+        pull_secret = json.load(pull_secret_fo)
+
+    # find all auths which might be related to the specified image
+    tmp_auths = []
+    for auth in pull_secret['auths']:
+        if auth in image:
+            tmp_auths.append(auth)
+    # get the most specific auth for particular image
+    tmp_auths = sorted(tmp_auths, key=len, reverse=True)
+    if tmp_auths:
+        # if there is match to particular auth, prepare authfile just with the
+        # matching auth
+        auth = tmp_auths[0]
+        # as key use only server name, without namespace
+        authfile_content = {
+            'auths': {
+                auth.split('/', 1)[0]: pull_secret['auths'][auth]
+            }
+        }
+    else:
+        # else use whole pull-secret
+        authfile_content = pull_secret
+
+    # create temporary auth file
+    with NamedTemporaryFile(mode='w', prefix='authfile_') as authfile_fo:
+        json.dump(authfile_content, authfile_fo)
+        # ensure the content will be saved into the file
+        authfile_fo.flush()
+        # pull original image (to be able to inspect it)
+        exec_cmd(f'podman image pull {image} --authfile {authfile_fo.name}')
+        # inspect the image and get full image url with tag
+        cmd_result = exec_cmd(f'podman image inspect {image}')
+        image_inspect = json.loads(cmd_result.stdout)
+        # if there is any tag specified, use it in the full image url,
+        # otherwise use url with digest
+        if image_inspect[0]['RepoTags']:
```

Comment:
Does each image contain this RepoTags definition? Or will it be safer to use .get('RepoTags') ?

Reply:
I was dealing with image, where `RepoTags` was set to None, but you are right, it will be safer to use `.get(...)`.

Reply:
Fixed in 43884ad.